Fix issue with ABC-derived classes having their members ignored in Python 3

jairideout opened the pull request:
ABC-derived classes (i.e. those deriving from the `abc` and `collections.abc` modules) now include their members (as defined in `__dict__`) in generated `autosummary` RST. `__slots__` may still be used to override `__dict__`, but if an ABC-derived class has empty slots, its `__dict__` will be used instead.

This was only an issue in Python 3 when ``:inherited-members:`` was not in effect.

Fixes #52

astrofrog commented:
@jairideout - thanks for the PR! It looks like tests are failing on Python 2.7 - since we are still Python 2 compatible (for now), could you see if there is an easy fix?

jairideout commented:
Thanks for the heads up @astrofrog -- I amended my commit to add Python 2.7 support and the tests are now passing. I'm not sure what's going on with the drop in coverage (-0.9%). Let me know if that's something I should investigate further.

Note: while adding Python 2.7 support, I noticed that the bug was only an issue in Python 3 when `:inherited-members:` was not in effect. I updated the relevant docs/comments in the PR and added a note about this to the corresponding issue (#52).

Thanks for taking a look!
